plugins/webapi/value/sendtransactionbyjson/handler.go

```diff
@@ -0,0 +1,238 @@
+	// ErrMalformedSignature defines a malformed signature error.
+	ErrMalformedSignature = fmt.Errorf("malformed signature")
+	// ErrWrongSignature defines a wrong signature error.
+	ErrWrongSignature = fmt.Errorf("wropng signature")
+	// ErrSignatureVersion defines a unsupported signature version error.
+	ErrSignatureVersion = fmt.Errorf("unsupported signature version")
+)
+
+// Handler sends a transaction.
+func Handler(c echo.Context) error {
+	sendTxMu.Lock()
+	defer sendTxMu.Unlock()
+
+	var request Request
+	if err := c.Bind(&request); err != nil {
+		return c.JSON(http.StatusBadRequest, Response{Error: err.Error()})
+	}
+
+	tx, err := NewTransactionFromJSON(request)
+	if err != nil {
+		return c.JSON(http.StatusBadRequest, Response{Error: err.Error()})
+	}
+
+	// validate transaction
+	err = valuetransfers.Tangle().ValidateTransactionToAttach(tx)
+	if err != nil {
+		return c.JSON(http.StatusBadRequest, Response{Error: err.Error()})
+	}
+
+	// Prepare value payload and send the message to tangle
+	payload, err := valuetransfers.ValueObjectFactory().IssueTransaction(tx)
+	if err != nil {
+		return c.JSON(http.StatusBadRequest, Response{Error: err.Error()})
+	}
+	_, err = issuer.IssuePayload(payload)
+	if err != nil {
+		return c.JSON(http.StatusBadRequest, Response{Error: err.Error()})
+	}
+
+	if err := valuetransfers.AwaitTransactionToBeBooked(tx.ID(), maxBookedAwaitTime); err != nil {
+		return c.JSON(http.StatusBadRequest, Response{Error: err.Error()})
+	}
+	return c.JSON(http.StatusOK, Response{TransactionID: tx.ID().String()})
+}
+
+// NewTransactionFromJSON returns a new transaction from a given JSON request or an error.
+func NewTransactionFromJSON(request Request) (*transaction.Transaction, error) {
+	// prepare inputs
+	inputs := make([]transaction.OutputID, len(request.Inputs))
+	for i, input := range request.Inputs {
+		b, err := base58.Decode(input)
+		if err != nil || len(b) != transaction.OutputIDLength {
+			return nil, ErrMalformedInputs
+		}
+		copy(inputs[i][:], b)
+	}
+
+	// prepare ouputs
+	outputs := make(map[address.Address][]*balance.Balance)
+	for _, output := range request.Outputs {
+		address, err := address.FromBase58(output.Address)
+		if err != nil {
+			return nil, ErrMalformedOutputs
+		}
+
+		balances := []*balance.Balance{}
+		for _, b := range output.Balances {
+			var color balance.Color
+			if b.Color == "IOTA" {
+				color = balance.ColorIOTA
+			} else {
+				colorBytes, err := base58.Decode(b.Color)
+				if err != nil || len(colorBytes) != balance.ColorLength {
+					return nil, ErrMalformedColor
+				}
+				copy(color[:], colorBytes)
+			}
+			balances = append(balances, &balance.Balance{
+				Value: b.Value,
+				Color: color,
+			})
+		}
+
+		outputs[address] = balances
+	}
+
+	// prepare transaction
+	tx := transaction.New(transaction.NewInputs(inputs...), transaction.NewOutputs(outputs))
+
+	// add data payload
+	if request.Data != "" {
+		data, err := base58.Decode(request.Data)
```

Comment:
Let's not overdo it with the base58 usage. Base58 is really ill-suited for long data as it requires BigInteger math over the entire data. The `Data` field should just be a `[]byte` then go will automatically decode/encode this as base64 which is much better for general data.